File: game.py
import pygame
import sys

# Initialize pygame
pygame.init()

# Screen dimensions
WIDTH, HEIGHT = 400, 600

# Colors
WHITE = (255, 255, 255)
BLACK = (0, 0, 0)
GOLD = (255, 215, 0)

# Font
FONT = pygame.font.Font(None, 36)

# Define thresholds for ranks
RANKS = [
    ("Bronze", 0),
    ("Silver", 45),
    ("Platinum", 65),
    ("Diamond", 100)
]

# Initialize screen
screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption("Coin Clicker Game")

# Load coin image and resize
coin_image = pygame.image.load("coin.png")  # Update with the correct path to your image
coin_image = pygame.transform.scale(coin_image, (100, 100))  # Resize the image to 100x100 pixels
coin_rect = coin_image.get_rect(center=(WIDTH // 2, HEIGHT // 2))

# Score and rank
score = 0

def get_rank(score):
    for rank, threshold in reversed(RANKS):
        if score >= threshold:
            return rank
    return "Bronze"

def draw_text(text, font, color, surface, x, y):
    textobj = font.render(text, True, color)
    textrect = textobj.get_rect(center=(x, y))
    surface.blit(textobj, textrect)

# Game loop
running = True
while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            pygame.quit()
            sys.exit()
        if event.type == pygame.MOUSEBUTTONDOWN:
            if coin_rect.collidepoint(event.pos):
                score += 1

    # Fill the screen with white
    screen.fill(WHITE)

    # Draw the coin
    screen.blit(coin_image, coin_rect)

    # Draw the score and rank
    rank = get_rank(score)
    draw_text(f"Score: {score}", FONT, BLACK, screen, WIDTH // 2, 50)
    draw_text(f"Rank: {rank}", FONT, GOLD, screen, WIDTH // 2, 100)

    # Update the display
    pygame.display.flip()

File: game.js
// Game constants
const WIDTH = 400;
const HEIGHT = 600;
const WHITE = '#ffffff';
const BLACK = '#000000';
const GOLD = '#ffd700';

// Initialize canvas and context
const canvas = document.getElementById('gameCanvas');
const ctx = canvas.getContext('2d');

// Load coin image
const coinImg = new Image();
coinImg.src = 'coin.png'; // Update with your coin image path

// Initialize game variables
let score = 0;
let coinRect = { x: WIDTH / 2 - 50, y: HEIGHT / 2 - 50, width: 100, height: 100 };

// Function to draw text on canvas
function drawText(text, x, y, color, font = '36px Arial') {
    ctx.fillStyle = color;
    ctx.font = font;
    ctx.fillText(text, x, y);
}

// Function to handle mouse click
function handleClick(event) {
    const rect = canvas.getBoundingClientRect();
    const mouseX = event.clientX - rect.left;
    const mouseY = event.clientY - rect.top;

    // Check if click is inside the coin area
    if (mouseX >= coinRect.x && mouseX <= coinRect.x + coinRect.width &&
        mouseY >= coinRect.y && mouseY <= coinRect.y + coinRect.height) {
        score++;
    }
}

// Add event listener for mouse click
canvas.addEventListener('click', handleClick);

// Game loop function
function gameLoop() {
    // Clear canvas
    ctx.clearRect(0, 0, WIDTH, HEIGHT);

    // Draw coin image
    ctx.drawImage(coinImg, coinRect.x, coinRect.y, coinRect.width, coinRect.height);

    // Draw score and rank
    drawText(`Score: ${score}`, WIDTH / 2 - 50, 50, BLACK);
    drawText(`Rank: ${getRank(score)}`, WIDTH / 2 - 50, 100, GOLD);

    // Request next frame
    requestAnimationFrame(gameLoop);
}

// Start the game loop
gameLoop();

// Function to determine rank based on score
function getRank(score) {
    const RANKS = [
        { name: 'Bronze', threshold: 0 },
        { name: 'Silver', threshold: 45 },
        { name: 'Platinum', threshold: 65 },
        { name: 'Diamond', threshold: 100 }
    ];

    for (let i = RANKS.length - 1; i >= 0; i--) {
        if (score >= RANKS[i].threshold) {
            return RANKS[i].name;
        }
    }

    return 'Bronze';
}
